src/treeffuser/_score_models/_neural_score_model.py

```diff
@@ -48,6 +83,41 @@ def forward(self, x: Float[t.Tensor, "batch x_dim"]) -> Float[t.Tensor, "batch y
         return self.model(x)
 
 
+class _CardLikeMLPModule(nn.Module):
+    def __init__(self, n_layers: int, hidden_size: int, input_size: int, output_size: int):
+        """
+        Simple MLP model with ReLU activation functions. Using Card like architecture.
+        In particular, this assumes that the input is of the form [y, x, t]
+        and an embedding of t is added after each layer.
+        """
+        super().__init__()
+        layers = []
+        t_embeddings = []
+
+        layers.append(nn.Linear(input_size - 1, hidden_size))
+        t_embeddings.append(nn.Linear(1, hidden_size))
+
+        for _ in range(n_layers - 1):
+            layers.append(nn.Linear(hidden_size, hidden_size))
+            t_embeddings.append(nn.Linear(1, hidden_size))
+
+        layers.append(nn.Linear(hidden_size, output_size))
+
+        self.layers = nn.ModuleList(layers)
+        self.t_embeddings = nn.ModuleList(t_embeddings)
+
+    def forward(self, x: Float[t.Tensor, "batch x_dim"]) -> Float[t.Tensor, "batch y_dim"]:
+        yx = x[:, :-1]
+        t = x[:, -1].unsqueeze(-1)
+        out = yx
+        for i, layer in enumerate(self.layers):
+            out = layer(out)
+            if i < len(self.layers) - 1:
+                out += self.t_embeddings[i](t)
+                out = nn.ReLU()(out)
```

Comment:
Technically CARD uses softplus ... but CARD also uses a parametrization for the mean so we are not exactly the same anyway :)

Reply:
Yes but I tried it and got worse results. By a lot. I could change this tho.